### PR TITLE
don't highlight literal identifiers anymore

### PR DESCRIPTION
**Pull request by fdietze**

### CODE REVIEW

File: tool-support/src/gedit/scala.lang [hunk 1]
```diff
@@ -159,6 +164,9 @@
       <keyword>Iterator</keyword>
       <keyword>Array</keyword>
       <keyword>String</keyword>
+      <keyword>IterableProxy</keyword>
```
Comment (jsuereth) [inline tool-support/src/gedit/scala.lang:167]: This doesn't look like literal-identifies :)

**Comment by jsuereth** [comment]
Just curious on why you want to stop highlighting these....

File: tool-support/src/gedit/scala.lang (hunk 1 above)
Comment (fdietze) [inline tool-support/src/gedit/scala.lang:167]: The second line of the commit message states that I also added the collection proxy types.

**Comment by fdietze** [comment]
Stopping the highlighting makes sense for statements like these:

val `type` = 5

of course this is a bad style, but still valid Scala code.

**Comment by jsuereth** [comment]
Right.  I'm asking _Why_ do you not want it highlighted?  It's nice to know when someone is using that style syntax, the same way using a `var` highlighted in the IDE is useful.

If you have a good reason to avoid highlighting those, great.  I'll pull it.

**Comment by fdietze** [comment]
An example where you have to use such a syntax are methods of a Java library that are named like keywords in Scala. Quote from Programming in Scala: "A typical use case is accessing the static yield method in Java’s Thread class. You cannot write Thread.yield() because yield is a reserved word in Scala. However, you can still name the method in back ticks, e.g., Thread.`yield`()."

**Comment by jsuereth** [comment]
I understand.  Why do we _not_ want to highlight this usage though?

**Comment by fdietze** [comment]
I think if you highlight something normal in a strange way it is annoying while reading the code.
On the other hand, the highlighter should allow you to use any style you want and just verify the correct syntax. Noticing bad style is the job of the IDE.

**Comment by jsuereth** [comment]
That's fine.  ` isn't bad style, but I do like to know when it has to be used.   I could go either way here in terms of highlighting or not.   Right now I like visually distinguishing these methods, since sometimes they can cause issues integrating with Java.

**Comment by fdietze** [comment]
I just had a look in Eclipse and tested some highlighting stuff.

It turns out that Eclipse also ignores the highlighting of keywords in backticks.

So, how should we proceed?

**Comment by jsuereth** [comment]
miming eclipse is a good enough reason for me.   Thanks for the work and the debate.

**Pull request merged**